### PR TITLE
IR-694 Initial AREAS table mapping

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/UserRole.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 @SuppressWarnings("unused")
 @Schema(description = "User Role")
 @JsonInclude(JsonInclude.Include.NON_NULL)
-@Builder
+@Builder(toBuilder = true)
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/UserRole.java:28]: I couldn't get it compile without this set

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/UserCaseloadRole.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public UserRole transform() {
         return builderForUserRole().build();
     }
 
-    private UserRoleBuilder builderForUserRole() {
+    private UserRole.UserRoleBuilder builderForUserRole() {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/UserCaseloadRole.java:49]: unnecessary change

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/GeographicRegion.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.hmpps.prison.repository.jpa.model;
+
+import jakarta.persistence.DiscriminatorValue;
+import jakarta.persistence.Entity;
+import lombok.NoArgsConstructor;
+
+@Entity
+@DiscriminatorValue(GeographicRegion.GEOGRAPHIC)
+@NoArgsConstructor
+public class GeographicRegion extends ReferenceCode {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/GeographicRegion.java:10]: new java class 😱

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/GeographicRegion.java (hunk 3 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/GeographicRegion.java:10]: well ideally any new files should be kotlin unless there is a reason for them to be java

Comment (petergphillips) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/GeographicRegion.java:10]: since we are gradually converting all to kotlin anyway

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/GeographicRegion.java:10]: I'll do it in the next commit
